kroxylicious-docs-validator/src/test/java/io/kroxylicious/doctools/validator/YamlCodeblockValidationDT.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.doctools.validator;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.asciidoctor.ast.StructuralNode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import io.kroxylicious.doctools.asciidoc.Block;
+import io.kroxylicious.doctools.asciidoc.BlockExtractor;
+import org.yaml.snakeyaml.Yaml;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+/** Tests that all yaml source blocks within the documentation are valid yaml */
```

Comment:
This is a replacement for extract-yaml-fencedcodeblocks.sh.  The intention is that it is functionally the same, with one difference.  If there was a Yaml source block with a AsciiDoc variable, it would be resolved prior to validation.

There's scope here to have more funky validations.

We could have schema validations for instance, by adding a schema ref to the `[source,yaml,schema="XXXX"]` or perhaps validations that the yaml is a valid CR `[source,yaml,crd="virtualkafkaclusters.kroxylicious.io"]`